pyuvdata/uvcal/initializers.py

```diff
@@ -0,0 +1,542 @@
+"""From-memory initializers for UVCal objects."""
+from __future__ import annotations
+
+from typing import Literal
+
+import numpy as np
+from astropy.coordinates import EarthLocation
+from astropy.time import Time
+
+from .. import __version__
+from ..docstrings import combine_docstrings
+from ..uvdata.initializers import (
+    XORIENTMAP,
+    Locations,
+    get_antenna_params,
+    get_freq_params,
+    get_spw_params,
+    get_time_params,
+)
+
+
+def new_uvcal(
+    *,
+    time_array: np.ndarray,
+    antenna_positions: np.ndarray | dict[str | int, np.ndarray],
+    telescope_location: Locations,
+    telescope_name: str,
+    cal_style: Literal["sky", "redundant"],
+    gain_convention: Literal["divide", "multiply"],
+    jones_array: np.ndarray | str,
+    x_orientation: Literal["east", "north", "e", "n", "ew", "ns"],
+    freq_array: np.ndarray | None = None,
+    freq_range: np.ndarray | None = None,
+    cal_type: Literal["delay", "gain"] | None = None,
+    integration_time: float | np.ndarray | None = None,
+    channel_width: float | np.ndarray | None = None,
+    antenna_names: list[str] | None = None,
+    antenna_numbers: list[int] | None = None,
+    antname_format: str = "{0:03d}",
+    ant_array: np.ndarray | None = None,
+    flex_spw_id_array: np.ndarray | None = None,
+    ref_antenna_name: str | None = None,
+    sky_catalog: str | None = None,
+    sky_field: str | None = None,
+    empty: bool = False,
+    data: dict[str, np.ndarray] | None = None,
+    history: str = "",
+    **kwargs,
```

Comment:
I don't see any documentation for this `kwargs`. Not sure if this should be done with your new parsing infrastructure or not.

Reply:
All the kwargs are added to the `Other Parameters` section, so they should be there. It doesn't explicitly mention the kwargs, per se.

Reply:
But there's no ` @combine_docstrings` decorator on this method, so I don't think anything will be added?

Reply:
Oh! I see what you mean. I'll check that out.